pkg/docker/config/config.go

```diff
@@ -136,7 +136,15 @@ func getAuthFilePaths(sys *types.SystemContext) []authPath {
 		// Logging the error as a warning instead and moving on to pulling the image
 		logrus.Warnf("%v: Trying to pull image in the event that it is a public image.", err)
 	}
+
+	var xdgCfgHomePath string
+	if xdgCfgHome := os.Getenv("XDG_CONFIG_HOME"); xdgCfgHome != "" {
+		xdgCfgHomePath = filepath.Join(xdgCfgHome, xdgRuntimeDirPath)
+	} else {
+		xdgCfgHomePath = filepath.Join(homedir.Get(), nonLinuxAuthFilePath)
+	}
```

Comment:
could a non-root user set XDG_CONFIG_HOME to somewhere they shouldn't?  Should we use securjoin for thie first one?

Reply:
The same user can already specify `--authfile` pointing anywhere else (unless the user somehow able to manipulate the environment of a future `su`/`sudo`, but not the command being run… I’m not too worried about that, and setting `XDG_RUNTIME_DIR` would allow the same attack already).

The suffix is a known constant, we don’t need `securejoin` to protect against that constant containing `../`.